### ISSUE TITLE
AADSTS500011: The resource principal named https://vault.microsoftazure.us was not found in the tenant named "MyUsGov GCC"...

### ISSUE THREAD

**Issue by Sean-Steefel** [hashicorp/terraform-provider-azurerm#16496]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.8

### AzureRM Provider Version

3.0.2

### Affected Resource(s)/Data Source(s)

data.azurerm_key_vault_secret

### Terraform Configuration Files

```hcl
terraform {
    required_providers {
        azurerm = {
            source = "hashicorp/azurerm"
            version = "=3.0.2"
        }
        azuread = {
            source = "hashicorp/azuread"
            version = "=2.20.0"
        }
    }
    backend "azurerm" {
        environment = "usgovernment"
        resource_group_name  = "fdo-rg-general-gcc-qat"
        storage_account_name = "fdostrgeneralgccqat"
        container_name       = "hfm"
        key                  = "tf/terraform_debug.tfstate"
    }
}

provider "azurerm" {
    environment = "usgovernment"
    features {
        key_vault {
            recover_soft_deleted_key_vaults=false
            purge_soft_delete_on_destroy=true
        }
    }
}

provider "azuread" {
    environment = "usgovernment"
}

data "azurerm_client_config" "current" {}

variable "azure_location"                       { default="usgovvirginia" }
variable "buildid"                              { default="12121" }
variable "CommitId"                             { default="222333111" }
variable "azure_environment_setting"            { default = "gcc-qat" }
variable "generalResourceKey"                   { default = "general" }
variable "commonResourceKey"                    { default = "common" }
variable "common_resource_name"                 {
                                                    default = "me"
                                                    description = "Common use name value for standard naming conventions"
                                                }

locals {
    resource_grp_name_common     = "fdo-rg-${var.commonResourceKey}-${var.azure_environment_setting}"
    resource_grp_name_general    = "fdo-rg-${var.generalResourceKey}-${var.azure_environment_setting}"
    current_user_object_id       = data.azurerm_client_config.current.object_id 
    current_user_tenant_id       = data.azurerm_client_config.current.tenant_id 
    current_user_subscription_id = data.azurerm_client_config.current.subscription_id
    azurerm_policies             = {
        apim = {
            tenant_id = local.current_user_tenant_id
            object_id = data.azurerm_api_management.apim.identity.0.principal_id
            key_permissions = [ 
            ]
            secret_permissions = [
                "Get"
            ]
            certificate_permissions = [
                "Get",
                "List",
            ]
        }
    }
    azurerm_secrets              = {
        apimAppId = {
            name  = "ClientId"
            value = data.azurerm_key_vault_secret.apimAppId.value
        }
        apimApiKey = {
            name  = "ClientSecret"
            value = data.azurerm_key_vault_secret.apimApiKey.value
        }
        apimAppResourceId = {
            name  = "apimAppResourceId"
            value = data.azurerm_key_vault_secret.apimAppResourceId.value
        }
        arsApimSubscriptionKey = {
            name  = "arsApimSubscriptionKey"
            value = data.azurerm_key_vault_secret.apimSubscriptionKey.value
        }
    }
}

data "azurerm_key_vault" "generalKeyvault" {
  name                = "fdo-kv-${var.generalResourceKey}-${var.azure_environment_setting}"
  resource_group_name = local.resource_grp_name_general
}

data "azurerm_key_vault_secret" "apimApiKey" {
  name         = "MEClientSecretForApim"
  key_vault_id = data.azurerm_key_vault.generalKeyvault.id
}

data "azurerm_key_vault_secret" "apimAppId" {
  name         = "meClientID"
  key_vault_id = data.azurerm_key_vault.generalKeyvault.id
}

data "azurerm_key_vault_secret" "apimAppResourceId" {
  name         = "apimAppId"
  key_vault_id = data.azurerm_key_vault.generalKeyvault.id
}

data "azurerm_key_vault_secret" "apimSubscriptionKey" {
  name         = "apimSubscriptionKey"
  key_vault_id = data.azurerm_key_vault.generalKeyvault.id
}

data "azurerm_api_management" "apim" {
  name                = "fdo-apim-${var.commonResourceKey}-${var.azure_environment_setting}"
  resource_group_name = local.resource_grp_name_common
}

# create resource group 
 resource "azurerm_resource_group" "rg" {
     name     = "fdo-rg-${var.common_resource_name}-${var.azure_environment_setting}"
     location = var.azure_location
     tags     = {
          environment = var.azure_environment_setting
          terraform   = "true"
          buildid     = var.buildid
          CommitId    = var.CommitId
     }
}
# create key vault for ingress of secrets
resource "azurerm_key_vault" "kv" {
     name                          = "fdo-kv-${var.common_resource_name}-${var.azure_environment_setting}"
     resource_group_name           = azurerm_resource_group.rg.name
     location                      = azurerm_resource_group.rg.location
     enabled_for_disk_encryption   = true
     enabled_for_deployment        = true
     tenant_id                     = data.azurerm_client_config.current.tenant_id 
     sku_name                      = "standard"
     
     tags = {
        environment      = var.azure_environment_setting
        terraform        = "true"
        buildid          = var.buildid
        CommitId         = var.CommitId
    }

    lifecycle {
        ignore_changes = [tags]
    }
}

# set key vault policies
resource "azurerm_key_vault_access_policy" "policies" {
     for_each = local.azurerm_policies
     key_vault_id = azurerm_key_vault.kv.id
     tenant_id = each.value["tenant_id"]
     object_id = each.value["object_id"]
     certificate_permissions = each.value["certificate_permissions"]
     key_permissions = each.value["key_permissions"]
     secret_permissions = each.value["secret_permissions"]
}

# set secret values
resource "azurerm_key_vault_secret" "azurerm_secrets" {
     for_each = local.azurerm_secrets
     key_vault_id = azurerm_key_vault.kv.id
     name = each.value["name"]
     value = each.value["value"]
}
```

### Debug Output/Panic Output

```shell
GCC based logs generated but compliance cannot allow upload of logs please reach out to me for assessment.
```

### Expected Behaviour

Vault secrets should have been transferred from the "General" key vault to the "Common" key vault

### Actual Behaviour

AAD Claim uri returned in the error indicates that a bug exists in accessing vault secrets the correct claim would be: https://vault.usgovcloudapi.net
![image](https://user-images.githubusercontent.com/73724562/164524743-02eaa030-c11c-4bae-be54-83084d9b7daf.png)

### Steps to Reproduce

terraform plan

### Important Factoids

Running in AzureUSGovernment

**Comment by Sean-Steefel** [comment]
Additional note: This was working as expected on AzureAD 2.8.0 and AzureRM 2.67.0
![image](https://user-images.githubusercontent.com/73724562/164523859-2433182b-24ad-4695-b30f-c85a66e78bb2.png)

**Comment by tombuildsstuff** [comment]
Duplicate of #16291 which was fixed in v3.3.0 of the Azure Provider
